spec/features/user_interact_with_queue_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "spec_helper"
+
+feature "User interacts with the queue" do
+  scenario "user adds and reorders videos in the queue" do
+    comedies = Fabricate(:category)
+    monk = Fabricate(:video, title: "Monk", category: comedies)
+    south_park = Fabricate(:video, title: "Sourth Park", category: comedies)
+    futurama = Fabricate(:video, title: "futurama", category: comedies)
+    sign_in
+
+    add_video_to_queue(monk)
+    expect_video_to_be_in_queue(monk)
+
+    visit video_path(monk)
+    expect_link_not_to_be_seen("+ My Queue")
+
+    add_video_to_queue(south_park)
+    add_video_to_queue(futurama)
+
+    set_video_position(monk, 3)
+    set_video_position(south_park, 1)
+    set_video_position(futurama, 2)
+
+    update_queue
+
+    expect_video_position(south_park, 1)
+    expect_video_position(futurama, 2)
+    expect_video_position(monk, 3)
+  end
+
+  def update_queue
+    click_button("Update Instant Queue")
+  end
+
+  def expect_video_to_be_in_queue(video)
+    page.should have_content(video.title)
+  end
+
+  def expect_link_not_to_be_seen(link_text)
+    page.should_not have_content(link_text)
+  end
+
+  def add_video_to_queue(video)
+    visit home_path
+    find("a[href='/videos/#{video.id}']").click
+    click_link "+ My Queue"
+  end
+
+  def set_video_position(video, position)
+    within(:xpath, "//tr[contains(.,'#{video.title}')]") do
+      fill_in "queue_items[][position]", with: position
+    end
+  end
+
+  def expect_video_position(video, position)
+    expect(find(:xpath, "//tr[contains(.,'#{video.title}')]//input[@type='text']").value).to eq(position.to_s)
```

Comment:
My only concern in this file is that you use `expect` RSpec syntax, and you also use `should` RSpec syntax - these are 2 different styles and it looks strange to me. I would recommend to stick with 1 style in whole project. Either one looks good, but just 1.